### PR TITLE
#85 move circle to ellipse

### PR DESCRIPTION
**Pull request by ensemenova**
Please review once again

### CODE REVIEW

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+    console.log(1)
+  }
+
+  if (current.y < bottomRightY.y && (firstQuarter || secondQuarter)){
+    topLeftY.y = current.y
+    console.log(2)
+  }
+  if (current.x < bottomRightX.x && (secondQuarter || thirdQuarter)){
+    topLeftX.x = current.x
+    console.log(3)
```
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:225]: please remove console.log calls

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+  const firstQuarter = current.x > topLeftX.x + rx && current.y <= topLeftY.y + ry
+  const secondQuarter =
+    current.x <= topLeftX.x + rx && current.y <= topLeftY.y + ry
+  const thirdQuarter = current.x <= topLeftX.x + rx && current.y > topLeftY.y + ry
+  const forthQuarter = current.x > topLeftX.x + rx && current.y > topLeftY.y + ry
+  console.log(rad, firstQuarter, secondQuarter, thirdQuarter, forthQuarter)
+  console.log(current, topLeftX,topLeftY, bottomRightX, bottomRightY)
+  if (firstQuarter) {
+
+  }
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:213]: empty if-statement

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+  const bottomRightY =
+    item.pos[0].y <= item.pos[1].y ? item.pos[1] : item.pos[0]
+  //check in which quarter the anchor is placed
+  const firstQuarter = current.x > topLeftX.x + rx && current.y <= topLeftY.y + ry
+  const secondQuarter =
+    current.x <= topLeftX.x + rx && current.y <= topLeftY.y + ry
+  const thirdQuarter = current.x <= topLeftX.x + rx && current.y > topLeftY.y + ry
+  const forthQuarter = current.x > topLeftX.x + rx && current.y > topLeftY.y + ry
+  console.log(rad, firstQuarter, secondQuarter, thirdQuarter, forthQuarter)
+  console.log(current, topLeftX,topLeftY, bottomRightX, bottomRightY)
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:210]: remove console.log

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+        item.pos[1].y = current.y
+        this.data.current = previousPos1
+      }
+    } else if (item.mode === SimpleObjectMode.line && anchor) {
+      const previousPos1 = anchor.get_xy0()
+      anchor.x = current.x
+      anchor.y = current.y
+      this.data.current = previousPos1
+    } else if (item.mode === SimpleObjectMode.rectangle && anchor) {
+      handleRectangleChangeWithAnchor.call(this, item, anchor, current)
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:307]: why not `handleRectangleChangeWithAnchor(item, anchor, current)` ? why do you need context here?

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+ * limitations under the License.
+ ***************************************************************************/
+import Vec2 from '../../util/vec2'
+import Base, { invalidateItem, OperationType } from './base'
+import { ReSimpleObject } from '../../render/restruct'
+import { SimpleObject, SimpleObjectMode } from 'src/script/chem/struct'
+import scale from "../../util/scale";
+import util from "../../render/util";
+
+const tfx = util.tfx
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:23]: what is 'tfx' ? what does it mean?

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+export function makeCircleFromEllipse(position0: Vec2, position1: Vec2): Vec2 {
+  const diff = Vec2.diff(position1, position0)
+  const min = Math.abs(diff.x) < Math.abs(diff.y) ? diff.x : diff.y
+  return new Vec2(
+    position0.x + (diff.x > 0 ? 1 : -1) * Math.abs(min),
+    position0.y + (diff.y > 0 ? 1 : -1) * Math.abs(min),
+    0
+  )
+}
```
Comment (TomasLight) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:337]: has no new line at the end of file

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 4 above)
Comment (AndreiMazol) [inline packages/ketcher-react/src/script/editor/operations/simpleObject.ts:307]: Why  do you use .call() ? I see handleRectangleChangeWithAnchor does not require any information  from SimpleObject instance, it works with input arguments only.

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 5 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts:23]: It alias for Float.toFixed, to compare more precise

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 4 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 1 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 6 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 4 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 2 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

File: packages/ketcher-react/src/script/editor/operations/simpleObject.ts (hunk 3 above)
Comment (ensemenova) [reply packages/ketcher-react/src/script/editor/operations/simpleObject.ts]: fixed

**Review by TomasLight — APPROVED** [review]

**Review by AndreiMazol — APPROVED** [review]

**Pull request merged**